editor/blocks/list/index.js

```diff
@@ -16,16 +16,56 @@ wp.blocks.registerBlock( 'core/list', {
 		)
 	},
 
-	edit( { attributes } ) {
-		const { listType = 'ol', items = [] } = attributes;
-		const value = items.map( item => {
+	controls: [
+		{
+			icon: 'editor-alignleft',
+			title: wp.i18n.__( 'Align left' ),
+			isActive: ( { align } ) => ! align || 'left' === align,
+			onClick( attributes, setAttributes ) {
+				setAttributes( { align: undefined } );
+			}
+		},
+		{
+			icon: 'editor-aligncenter',
+			title: wp.i18n.__( 'Align center' ),
+			isActive: ( { align } ) => 'center' === align,
+			onClick( attributes, setAttributes ) {
+				setAttributes( { align: 'center' } );
+			}
+		},
+		{
+			icon: 'editor-alignright',
+			title: wp.i18n.__( 'Align right' ),
+			isActive: ( { align } ) => 'right' === align,
+			onClick( attributes, setAttributes ) {
+				setAttributes( { align: 'right' } );
+			}
+		},
+		{
+			icon: 'editor-justify',
+			title: wp.i18n.__( 'Justify' ),
+			isActive: ( { align } ) => 'justify' === align,
+			onClick( attributes, setAttributes ) {
+				setAttributes( { align: 'justify' } );
+			}
+		}
+	],
+
+	edit( { attributes, setAttributes } ) {
+		const { listType = 'ol', items = [], align } = attributes;
+		const content = items.map( item => {
 			return `<li>${ item.value }</li>`;
 		} ).join( '' );
+		const defaultListStyles = { listStylePosition: 'inside' };
+		const alignment = align ? { textAlign: align } : {};
+		const style = { ...alignment, ...defaultListStyles };
 
 		return (
 			<Editable
 				tagName={ listType }
-				value={ value } />
+				onChange={ ( value ) => setAttributes( { content: value } ) }
```

Comment:
We need to think more about this `onChange` handler. The block's state should be reflected by its `items` attributes only. `content` is not really applicable in this case. What we'd likely want to do is infer the new set of items on the `onChange` handler and update attributes there. We can do that in a subsequent pull request.